eclass/linux-mod-r1.eclass

```diff
@@ -650,6 +650,28 @@ _modules_prepare_kernel() {
 	fi
 
 	linux-info_pkg_setup
+
+	if use dist-kernel &&
+		! has_version "~virtual/dist-kernel-${KV_MAJOR}.${KV_MINOR}.${KV_PATCH}"
+	then
+		ewarn
+		ewarn "The kernel modules in ${CATEGORY}/${PN} are being built for"
+		ewarn "kernel version ${KV_FULL}. But this does not match the"
+		ewarn "installed version of virtual/dist-kernel!"
+		ewarn
+		ewarn "The slot dependency metadata of the built package will not"
+		ewarn "accurately reflect the kernel version the kernel modules were"
+		ewarn "actually built for."
```

Comment:
Maybe this part is not necessary? It does explain some implications, but to normal users I think it'll just sound confusing.

Mismatching version is odd in its own right and likely not intentional in most cases without needing to explain further I think.

Reply:
> Mismatching version is odd in its own right and likely not intentional in most cases without needing to explain further I think.

This makes sense, dropped!

I also added a condition for when the KV_FULL does not end with `-dist`. If for example we have gentoo-sources eselected with coincidentally the correct version, then we still want to show this warning because the configuration is still "weird" for `USE=dist-kernel`

Reply:
>I also added a condition for when the KV_FULL does not end with -dist

That sounds annoying, people won't necessarily set that up with USE=savedconfig (I don't for one, so I'm gonna get the message)

Reply:
> >I also added a condition for when the KV_FULL does not end with -dist
> 
> That sounds annoying, people won't necessarily set that up with USE=savedconfig (I don't for one, so I'm gonna get the message)

Right, is there some other way we can check if the eselected kernel is a dist-kernel and not some other kernel source that happens to have the same version?

Reply:
None that I can think of, unless dist-kernel's end adds something to identify itself regardless (doesn't necessarily have to be in KV_FULL, e.g. an extra file in /usr/src/linux). But don't believe it's important enough to need to worry about this, many times people mix both sources and dist the versions end up mismatching before long, or the sources are left unbuilt and the modules will fail anyway.

Reply:
Okay I have dropped this new condition and left only the part where we check the kernel version, this is the most important part anyway.